src/__tests__/pages/teachers/programmes/[programmeSlug]/units/[unitSlug]/lessons/[lessonSlug].test.tsx

```diff
@@ -76,6 +90,24 @@ describe("pages/teachers/lessons", () => {
     expect(screen.queryByTestId("download-all-button")).not.toBeInTheDocument();
   });
 
+  it("share button is disabled with non legacy content", async () => {
+    render(
+      <LessonOverviewPage
+        curriculumData={lessonOverviewFixture({
+          hasDownloadableResources: false,
+          expired: true,
+        })}
+      />,
+    );
+
+    const shareButton = screen.queryAllByTestId("share-all-button");
+    const shareLabel = screen.queryAllByText("Share activities with pupils");
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
I think I'd prefer doing something like `if (shareButton.length === 0) throw new error('cant find any shre buttons) else ...`, so if there are no buttons and it throws an error it's super clear why